app/models/message_object_datum.rb

```diff
@@ -11,5 +11,6 @@
 class MessageObjectDatum < ApplicationRecord
   belongs_to :message_object
 
+  after_create_commit { message_object.message.set_html_visualization if message_object.form? }
```

Comment:
Tu by som este zvazil jednu vec, namiesto toho aby sa to udialo ako commit, update, update, tak by mozno davalo zmysel ten callback spravit tak, ze nevola naozaj update ale len to nastavi, potom save to ulozi v jednom/dvoch update.

ked sa vola nieco `set_` tak ocakavam, ze to nastavi, nie ze to ulozi do db.

Reply:
before_create_commit a v hooku len nastavit tie fieldy nestaci?

Reply:
To by bolo fajn, len tie updates sa deju na inych objektoch, na `message` a `message_object`e, lenze callback je na `message_object_datum`.
Premenovala som teda aspon tu metodu.

Reply:
To mi nepride ako problem, ze sa to deje na inom objekte, ale uznavam, ze neviem ako sa tu volaju tie callbacky.

Reply:
Ano, to sme riesili na calle, ze to nie je bezna situacia, ale bohuzial to tak je (hlavne kvoli importovanym draftom zo ZIPka, kde sa obsah podania nacitava postupne, v roznych joboch).
Vizualizaciu vieme vytvorit az vtedy, ked mame nacitany obsah formulara.